account_wallet/models/account_wallet.py

```diff
@@ -0,0 +1,116 @@
+# © 2015  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountWallet(models.Model):
+    _name = 'account.wallet'
+    _inherit = ['portal.mixin', 'mail.thread', 'mail.activity.mixin']
+    _description = "Wallet"
+    _check_company_auto = True
+
+    name = fields.Char(
+        readonly=True,
+        copy=False,
+        default=lambda self: _("New"),
+    )
+    wallet_type_id = fields.Many2one(
+        'account.wallet.type',
+        'Wallet Type',
+        required=True,
+        ondelete='restrict')
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Partner",
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        related="wallet_type_id.company_id",
+        store=True,
+    )
+    company_currency_id = fields.Many2one(
+        related='company_id.currency_id',
+        readonly=True,
+    )
+    balance = fields.Monetary(
+        compute='_compute_balance',
+        currency_field='company_currency_id',
+        store=True,
+        tracking=10
+    )
+    active = fields.Boolean(
+        default=True)
+    account_move_line_ids = fields.One2many(
+        comodel_name="account.move.line",
+        inverse_name="account_wallet_id",
+        string="Journal Items")
+    # TODO: Check if this is necessary
+    # create_date = fields.Date(
+    #    default=fields.Date.today)
+
+    _sql_constraints = [(
+        'wallet_uniq',
+        'EXCLUDE (partner_id WITH =, wallet_type_id WITH =, company_id WITH =) '
+        'WHERE (active=True)',
+        'You can have just one active wallet for same type and partner by company')]
+
+    @api.constrains('partner_id')
+    def _check_partner(self):
+        """ Check there is no move lines to be able to set a partner
+        """
+        for wallet in self:
+            if wallet.partner_id and wallet.account_move_line_ids:
+                raise ValidationError(_('Partner can not be defined on a'
+                                        ' cagnotte with journal items'))
+        return True
+
+    def _get_name(self):
+        """
+            Get a composed display name from different properties
+
+        """
+        self.ensure_one()
+        name = "{type} - {name}"
+        values = {
+            "type": self.wallet_type_id.name,
+            "name": self.name,
+        }
+        if self.partner_id.name:
+            name += " - {partner}"
+            values.update({
+                "partner": self.partner_id.name,
+            })
+        return name.format(**values)
+
+    def name_get(self):
```

Comment:
with a specific name_get maybe it's better to do the same with name_search

Reply:
Ok, I'll add it as TODO